### PR TITLE
Dynamo for revit ziyun

### PR DESCRIPTION
**Pull request by ZiyunShang**
This PR is mainly used to get the path of journal file which is running in process.

### CODE REVIEW

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 1]
```diff
@@ -84,6 +84,14 @@ private void SetDefaultValuesToUninitializedProperties()
         {
             var fi = new FileInfo(Assembly.GetExecutingAssembly().Location);
             string assDir = fi.DirectoryName;
+            
+            //It is used to get the path of journal file.
+            if (RevitSystemTestBase.IsJournalReplaying())
```
Comment (QilongTang) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: May I ask why are we doing the check here? I feel even if it is playing we can still get the path?

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 2]
```diff
@@ -84,6 +84,14 @@ private void SetDefaultValuesToUninitializedProperties()
...
+            {
+                string journalPath = RevitTestExecutive.CommandData.Application.Application.RecordingJournalFilename;
+                WorkingDirectory = journalPath.Substring(0, journalPath.LastIndexOf('\\'));
+                SamplesPath = journalPath.Substring(0, journalPath.LastIndexOf('\\'));
```
Comment (QilongTang) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: Maybe use `Path.GetDirectoryName()` is easier to understand

**Comment by QilongTang** [comment]
One comment, then LGTM

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 3]
```diff
@@ -190,6 +198,17 @@ protected override void SetupCore()
                 emptyModelPath1 = Path.Combine(workingDirectory, "empty1.rfa");
             }
         }
+        
+        //Indicates whether it is in journal replaying mode.
+        public static bool IsJournalReplaying()
```
Comment (mjkkirschner) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: what does it refer to here? Revit or the testbase class?
Can you use triple slash comments with summaries on public methods.

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs (hunk 1 above)
Comment (mjkkirschner) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: can you add a comment here on why these paths are set to this directory if a journal is running? What is this doing?

Comment (ZiyunShang) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: We put each type of journal file, such as AdaptiveComponent, AnalysisDisplay, and their Revit files and dynamo files into Revit's Regression \ D4R folder. The results are Regression \ D4R \ AdaptiveComponent, Regression \ D4R \ AnalysisDisplay, The path to the file is different. We have to get where each journal file runs when Revit gets it, and then get the location of the required Revit and dynamo files.

Comment (QilongTang) [reply test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: I think @mjkkirschner may mean that these comments added in the code?

**Comment by ZiyunShang** [comment]
@QilongTang @mjkkirschner 
Is there something else to improve?

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs (hunk 1 above)
Comment (mjkkirschner) [reply test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: @ZiyunShang yes, please add this comment or something similar to the source code.

**Comment by mjkkirschner** [comment]
this LGTM - do you intend to cherry pick this to master branch / and other revit years?

**Comment by ZiyunShang** [comment]
@mjkkirschner 
Just the branch RC1.3.3_Revit2019, please

**Comment by QilongTang** [comment]
LGTM

**Comment by mjkkirschner** [comment]
@QilongTang I think we'll need to merge this when ready.

**Comment by QilongTang** [comment]
@ZiyunShang @philxia Merged

**Pull request merged**